commit ee58992be3a716ee3862fe9dc0efea7406eac486
Author: Jan Hermann <dev@jan.hermann.name>
Date:   Fri Nov 26 14:18:59 2021 +0100

    increase threshold for gradient test

diff --git a/tests/mbd_grad_test_cases.f90 b/tests/mbd_grad_test_cases.f90
index d87816c..19a0b3f 100644
--- a/tests/mbd_grad_test_cases.f90
+++ b/tests/mbd_grad_test_cases.f90
@@ -269,7 +269,7 @@ subroutine test_mbd_ewald_deriv_expl()
     end do
     call geom%destroy()
     diff = (gradients - gradients_anl) / gradients_anl
-    if (failed(maxval(abs(diff)), 1d-8)) then
+    if (failed(maxval(abs(diff)), 2d-8)) then
         call print_matrix('delta gradients', diff)
         call print_matrix('anl', gradients_anl)
         call print_matrix('num', gradients)
